src/sgame/sg_cmds.cpp

```diff
@@ -3497,6 +3497,111 @@ void Cmd_Reload_f( gentity_t *ent )
 	}
 }
 
+/*
+=================
+Cmd_TeamStatus_f
+=================
+*/
+void Cmd_TeamStatus_f( gentity_t * ent )
+{
+	int builders = 0;
+	int arm = 0, mediboost = 0, miner = 0;
+	int omrccount = 0, omrchealth = 0;
+	bool omrcbuild = false;
+	gentity_t *tmp;
+
+	if ( !g_teamStatus.Get() ) 
+	{
+		trap_SendServerCommand( ent - g_entities, "print \"teamstatus is disabled.\n\"" );
+		return;
+	}
+
+	if ( ent->client->pers.namelog->muted ) 
+	{
+		trap_SendServerCommand( ent - g_entities, "print \"You are muted and cannot use message commands.\n\"" );
+		return;
+	}
+
+	if ( ent->client->pers.lastTeamStatus &&
+		( level.time - ent->client->pers.lastTeamStatus ) <
+	    g_teamStatus.Get() * 1000 ) 
+	{
+		trap_SendServerCommand( ent - g_entities, va( "print \"You may only check your team's status once every %i seconds.\"", g_teamStatus.Get() ) );
+		return;
+	}
+
+	ent->client->pers.lastTeamStatus = level.time;
+
+	tmp = &g_entities[ 0 ];
+	for ( int i = 0; i < level.num_entities; i++, tmp++ ) 
+	{
+		if ( i < MAX_CLIENTS ) 
+		{
+			if ( tmp->client &&
+			    tmp->client->pers.connected == CON_CONNECTED &&
+			    tmp->client->pers.team ==
+			    ent->client->pers.team && 
+				tmp->entity->Get<HealthComponent>()->Alive()
+			    && (tmp->client->ps.stats[ STAT_CLASS ] ==
+				PCL_ALIEN_BUILDER0
+				|| tmp->client->ps.stats[ STAT_CLASS ]  ==
+				PCL_ALIEN_BUILDER0_UPG
+				|| BG_InventoryContainsWeapon( WP_HBUILD, tmp->client->ps.stats ) ) )
+				builders++;
+			continue;
+		}
+
+		if ( tmp->s.eType == entityType_t::ET_BUILDABLE ) 
+		{
+			HealthComponent* health = tmp->entity->Get<HealthComponent>();
+			if ( tmp->buildableTeam != ent->client->pers.team || !health->Alive() )
+				continue;
+
+			switch ( tmp->s.modelindex ) 
+			{
+				case BA_H_REACTOR:
+				case BA_A_OVERMIND:
+					omrccount++;
+					if ( health && health->Health() > omrchealth )
+						omrchealth = health->Health();
+					if ( !omrcbuild )
+						omrcbuild = tmp->spawned;
+					break;
+				case BA_H_ARMOURY:
+					arm++;
+					break;
+				case BA_H_MEDISTAT:
+				case BA_A_BOOSTER:
+					mediboost++;
+					break;
+				case BA_H_DRILL:
+				case BA_A_LEECH:
+					miner++;
+					break;
+				default:
+					break;
+			}
+		}
+	}
+	
+	if ( ent->client->pers.team == TEAM_ALIENS ) 
```

Comment:
Note that you can get a buildable's "humanname" by using i.e. `BG_Buildable( type )->humanname`. I could be used to make code here more generic.